daemon/server/router/system/system_routes.go

```diff
@@ -197,40 +212,74 @@ func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter,
 		return err
 	}
 
-	var builderSize int64
-	if versions.LessThan(version, "1.42") {
-		for _, b := range buildCache {
-			builderSize += b.Size
+	var v system.DiskUsage
+	if systemDiskUsage != nil && systemDiskUsage.Images != nil {
+		v.ImageUsage = &system.ImageDiskUsage{
+			ActiveCount: systemDiskUsage.Images.ActiveCount,
+			Reclaimable: systemDiskUsage.Images.Reclaimable,
+			TotalCount:  systemDiskUsage.Images.TotalCount,
+			InUse:       systemDiskUsage.Images.TotalSize,
 		}
-	}
 
-	du := backend.DiskUsage{}
-	if getBuildCache {
-		du.BuildCache = &backend.BuildCacheDiskUsage{
```

Comment:
For API < v1.52 we can keep the old conversion code and return the old API struct; move the old API struct in the backend, and use it as response type for API <v1.52.